### DISCUSSION TITLE
Zksync Plugin - Error Msg: Solidity Verification Failed

### DISCUSSION THREAD

**Discussion by Ali757bey** [Cyfrin/foundry-full-course-cu#3005] [Q&A]
![Screenshot 2024-12-07 093335](https://github.com/user-attachments/assets/78ea4c14-feff-4ff8-85a7-902c95aa0a54)

My contract is deploying but verification is failing. I attempted to verify it on zksync sepolia explorer but I was getting an error message there as well stating "Deployed bytecode is not equal to generated one from given source"
![image](https://github.com/user-attachments/assets/2a1b670c-cb88-4198-acd7-93fe7d9ca0e1)

I didn't know if the plugin was using hardhat for verification or if it was pulling it from my local so I even attempted to change the node version to 18 because it's showing v22 is not supported. 

![image](https://github.com/user-attachments/assets/0ce6bcad-a2b1-498d-bbfd-28129e6756a7)

After changing node versions from 20 (never was on 22) to 18 and restarted my browser (Chrome) it's still showing the same error after verifying and deploying. 

Any suggestions or is the verification not that important just keep moving forward?

**Comment by Ali757bey** [comment]
I was working in a different workspace I created in remix named "Projects for Cyfrin" I seen this https://github.com/Cyfrin/foundry-full-course-cu/discussions/2726#discussioncomment-11025633 and after making the necessary changes it worked. 

Thank you
~Ali

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out.
